includes/Core/Site_Health/Tags_Placement.php

```diff
@@ -0,0 +1,257 @@
+				);
+
+				return $tests;
+			}
+		);
+	}
+
+	/**
+	 * Gets all REST routes.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return REST_Route[] List of REST_Route objects.
+	 */
+	private function get_rest_routes() {
+		return array(
+			new REST_Route(
+				'core/site/data/tags-placement-test',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => $this->get_method_proxy( 'tags_placement_test' ),
+						'permission_callback' => function () {
+							return current_user_can( Permissions::SETUP );
+						},
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Checks if the modules tags are placed on the website.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Site health status results.
+	 */
+	protected function tags_placement_test() {
+		global $wp_version;
+
+		$result = array(
+			'label'   => __( 'Tags Placement', 'google-site-kit' ),
+			'status'  => 'recommended',
+			'badge'   => array(
+				'label' => __( 'Site Kit', 'google-site-kit' ),
+				'color' => 'blue',
+			),
+			'actions' => '',
+			'test'    => 'tag_placement',
+		);
+
+		if ( version_compare( $wp_version, '5.6', '<' ) ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'This feature requires WordPress version 5.6 or higher', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
+		$active_modules = $this->get_active_modules();
+		if ( empty( $active_modules ) ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'Tag status not available: AdSense, Tag Manager, and Analytics modules are not connected.', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
+		// Generate random page name that will result in 404 page, to prevent receiving
+		// cached page and target page with smaller content.
+		$random_page = substr(
+			str_shuffle( '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ' ),
+			0,
+			10
+		);
+		$response    = wp_remote_get( site_url( $random_page ) ); // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get
+
+		if ( is_wp_error( $response ) ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'There was an error while trying to get the status, please try again later.', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
+		$response = wp_remote_retrieve_body( $response );
+
+		$description = array();
+		foreach ( $active_modules as $module ) {
+			$tag_found = $this->check_if_tag_exists( $module, $response );
+
+			if ( $tag_found ) {
+				$description[] = $tag_found;
+			}
+		}
+
+		if ( ! empty( $description ) ) {
+			$result['description'] = '<ul>' . join( "\n", $description ) . '</ul>';
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Gets active modules filtered to account only for
+	 * Analytics, AdSense and Tag Manager.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Filtered active modules instances.
+	 */
+	protected function get_active_modules() {
```

Comment:
The name of the method here is misleading. It would be better if it conveyed that it returns active modules with tags, otherwise this is easy to misinterpret elsewhere where it's used.